language/extensions/move-table-extension/sources/Table.move

```diff
@@ -0,0 +1,44 @@
+/// Type of large-scale storage tables.
+module Extensions::Table {
+    struct Table<phantom K, phantom V> has store {
+        handle: u128
+    }
+
+    /// Create a new Table.
+    public fun new<K, V>(): Table<K, V> {
+        Table{handle: new_table_handle()}
+    }
+
+    native fun new_table_handle(): u128;
+
+    /// Destroy a table. The table must be empty to succeed.
+    native fun destroy(table: Table);
```

Comment:
Maybe destroy_empty? I think we used that naming in Vector, but I'd have to double check. Whatever that name is, let's choose the name that is consistent with Vector's empty resource case

Reply:
Done